src/Spago/Bower.hs

```diff
@@ -87,23 +100,34 @@ mkPackageName spagoName = do
       pure name
 
 
+-- | If the given version exists in bower, return a shorthand bower
+-- | version, otherwise return a URL#version style bower version.
+mkBowerVersion :: Spago m => Bower.PackageName -> Text -> Text -> m Bower.VersionRange
+mkBowerVersion packageName version repo = do
+
+  let args = ["info", "--json", Bower.runPackageName packageName <> "#" <> version]
+  (code, stdout, stderr) <- runBower args
+
+  when (code /= ExitSuccess) $ do
+    die $ "Failed to run: `bower " <> Text.intercalate " " args <> "`\n" <> stderr
+
+  info <- case Aeson.decode $ encodeUtf8 $ fromStrict stdout of
+    Just (Object obj) -> pure obj
+    _ -> die $ "Unable to decode output from `bower " <> Text.intercalate " " args <> "`: " <> stdout
+
+  if HashMap.member "version" info
+    then pure $ Bower.VersionRange $ "^" <> version
+    else pure $ Bower.VersionRange $ repo <> "#" <> version
+
+
 mkDependencies :: Spago m => Config -> m [(Bower.PackageName, Bower.VersionRange)]
 mkDependencies config = do
-
-  reposMeta <- GlobalCache.getMetadata Nothing
   deps <- Packages.getDirectDeps config
-
   for deps $ \(PackageName{..}, Package{..}) -> do
```

Comment:
I was wondering if we should do this in parallel, as from what I've seen `bower info` can be quite slow. We could use [an async pool like we do here](https://github.com/spacchetti/spago/blob/3eaba105a0a9ff59a35709e5d6336c1d81d73215/src/Spago/FetchPackage.hs#L67), or just [`mapTasks`](http://hackage.haskell.org/package/async-pool-0.9.0.2/docs/Control-Concurrent-Async-Pool.html#v:mapTasks) as we don't need the fancy error handling

Reply:
(this is not a requirement for merging, just noting that it might a bottleneck on larger packages)

Reply:
Yeah - done. Bower info is pretty slow (it takes a couple of seconds per call here, but I do have very high latency internet).